internal/handlers/initrd.go

```diff
@@ -68,5 +69,10 @@ func (h *initrdHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 
 	fileName := fmt.Sprintf("%s-initrd.img", imageID)
 	w.Header().Set("Content-Disposition", fmt.Sprintf("attachment; filename=%s", fileName))
-	http.ServeContent(w, r, fileName, time.Now(), initrdReader)
+	modTime, err := http.ParseTime(lastModified)
+	if err != nil {
+		log.Warnf("Error parsing last modified time %s: %v", lastModified, err)
+		modTime = time.Now()
```

Comment:
Maybe worth adding a negative test for this, i.e. in case of an err in ParseTime.